### PR TITLE
Document anomalous src_port param values

### PR DESCRIPTION
**Pull request by ffoulkes**
- Added an explanatory comment to each instance where the `src_port` param is set to the value of the `dst_port` input.

- Replaced the `dst_port` temporary variable with a `src_port` variable.

- Adjusted the value of `src_port` when initializing it, rather than in the call to `EncodeBytes()`.

### CODE REVIEW

File: ovs-p4rt/sidecar/ovsp4rt.cc [hunk 1]
```diff
@@ -723,10 +723,13 @@ void PrepareVxlanEncapTableEntry(p4::v1::TableEntry* table_entry,
...
                                      ACTION_VXLAN_ENCAP_PARAM_SRC_PORT));
-      uint16_t dst_port = htons(tunnel_info.dst_port);
+      // The src_port param is an arbitrary value that has nothing to do
+      // with what was specified in the tunnel_info structure. This is a
+      // workaround for a bug in the Linux Networking P4 program.
+      uint16_t src_port = htons(tunnel_info.dst_port);
 
-      param->set_value(EncodeByteValue(2, (((dst_port * 2) >> 8) & 0xff),
-                                       ((dst_port * 2) & 0xff)));
+      param->set_value(EncodeByteValue(2, (((src_port * 2) >> 8) & 0xff),
```
[the remark this answers is not in the document]
Comment (ffoulkes) [reply ovs-p4rt/sidecar/ovsp4rt.cc:731]: This is not a change to the value being assigned. It is a change to the name of the temporary variable being used for the assignment.

As I noted in the CL comment, the temporary variable should reflect the name of the param being set (`src_port`).

[the remark this answers is not in the document]
Comment (n-sandeep) [reply ovs-p4rt/sidecar/ovsp4rt.cc:732]: This looks fine, yes. Till p4 program is fixed we just use a value for SRC_PORT which is 2 times of DEST_PORT. 
And this code should be removed once P4 program changes are made for both Geneve and VxLAN.

[the remark this answers is not in the document]
Comment (5abeel) [reply ovs-p4rt/sidecar/ovsp4rt.cc]: Approving based on @n-sandeep feedback and comment.

@kamalakannanr89 please add this to your backlog on cleanup needed in P4 program in the future

**Review by 5abeel — APPROVED** [review]
lgtm

File: ovs-p4rt/sidecar/ovsp4rt.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (ffoulkes) [reply ovs-p4rt/sidecar/ovsp4rt.cc]: With the benefit of hindsight, I've:

1. Moved the adjustment (doubling) of the value from the call to `EncodeBytes()` to the assignment statement, so the `src_port` temporary variable contains the value the intend to use.
2. Modified the call to `EncodeBytes()` to use the `src_port` variable as-is, without further modification.

This is a much clearer solution that I believe properly addresses Sabeel's objection.

**Review by 5abeel — APPROVED** [review]
LGTM

**Pull request merged**